### PR TITLE
Add config file and update karma handlers

### PR DESCRIPTION
**Pull request by Joeyh021**
also adds custom bot class for hooking stuff into bot state, such as karma store

### CODE REVIEW

File: config.example.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+owners: # user IDs of bot owners with admin perms
+  - 1234
+  - 5678
+
+# emoji that should be used for karma
+# either a unicode emoji, name of reaction, or emoji ID
+# mapped to karma value
+emoji:
+  😊: 1
+  1000161402704633956: -5
```
Comment (otwl169) [inline config.example.yaml:11]: What is this

**Review by otwl169 — APPROVED** [review]
Send me your config file so I can add my user ID

File: config.example.yaml (hunk 1 above)
Comment (Joeyh021) [reply config.example.yaml:11]: The config file has a mapping of emoji to value
emoji can be either a literal emoji, the discord emoji ID, or the name of the emoji

**Comment by Joeyh021** [comment]
Already added your ID, I copied it from the old array in the code

File: config.example.yaml (hunk 1 above)
Comment (otwl169) [reply config.example.yaml:11]: Ok add comment for what emoji like :smile:

Comment (otwl169) [reply config.example.yaml:11]: The :emoji:

Comment (Joeyh021) [reply config.example.yaml:11]: wdym

Comment (otwl169) [reply config.example.yaml:11]: What server emoji from and what emoji called like # :sus: Uni of ww discord

Comment (Joeyh021) [reply config.example.yaml:11]: oh nah these are just examples they're overwritten in the actual config, not meant to have actual emoji. That ID isnt even an emoji im pretty sure

Comment (otwl169) [reply config.example.yaml:11]: 👌😳

**Pull request merged**